core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1250,9 +1249,9 @@ protected Scope visitTable(Table table, Optional<Scope> scope)
             ImmutableMap.Builder<ColumnHandle, Field> fields = ImmutableMap.builder();
 
             // TODO: discover columns lazily based on where they are needed (to support connectors that can't enumerate all tables)
```

Comment:
@kokosing @Praveen2112 @kasiafi Any idea what does this comment mean? 
Maybe some examples?

Reply:
It could be a case that connector is not able to enumerate columns because they are dynamic. But if you ask for a column then it could generate it for you. 

I guess we don't have such connectors yet, and in case we would have them I guess there is more work than just addressing this comment.

Reply:
Thank you.